src/TraceEvent/TraceLog.cs

```diff
@@ -153,10 +153,227 @@ public static TraceLog OpenOrConvert(string etlOrEtlxFilePath, TraceLogOptions o
+        /// the .Log Property) which lets you get at aggregated information (Processes, threads, images loaded, and perhaps most
+        /// importantly TraceEvent.CallStack() will work. Thus you can get real time stacks from events).
+        /// </summary>
+        /// <param name="rundownSession">
+        /// If given, the rundownSession is used to initialize module and method information and then the session is closed.
+        /// This only makes sense in realtime sessions when you need to resolve function names.
+        /// </param>
+        /// <example>
+        /// var client = new DiagnosticsClient(Process.GetCurrentProcess().Id);
+        /// using var rundownSession = client.StartEventPipeSession(
+        ///     new EventPipeProvider(ClrTraceEventParser.ProviderName, EventLevel.Informational, (long) ClrTraceEventParser.Keywords.Default),
+        ///     requestRundown: true
+        /// );
+        /// EventPipeProvider[] providers = new[]
+        /// {   new EventPipeProvider(ClrTraceEventParser.ProviderName, EventLevel.Informational, (long) ClrTraceEventParser.Keywords.Default),
+        ///     new EventPipeProvider(SampleProfilerTraceEventParser.ProviderName, EventLevel.Informational),
+        /// };
+        /// var session = client.StartEventPipeSession(providers, requestRundown: false);
+        /// var eventSource = TraceLog.CreateFromEventPipeSession(session, rundownSession);
+        /// eventSource.Process();
+        /// </example>
+        public static TraceLogEventSource CreateFromEventPipeSession(EventPipeSession session, EventPipeSession rundownSession = null)
+        {
+            var tlEventSource = CreateFromEventPipeEventSource(new EventPipeEventSource(session.EventStream));
+
+            if (rundownSession != null)
+            {
+                tlEventSource.TraceLog.ProcessInitialRundown(rundownSession);
+            }
+
+            return tlEventSource;
+        }
+
+        /// <summary>
+        /// From a EventPipeEventSource, create a real time TraceLog Event Source. Like an EventPipeEventSource a TraceLogEventSource
+        /// will deliver events in real time. However an TraceLogEventSource has an underlying Tracelog (which you can access with
+        /// the .Log Property) which lets you get at aggregated information (Processes, threads, images loaded, and perhaps most
+        /// importantly TraceEvent.CallStack() will work. Thus you can get real time stacks from events).
+        /// </summary>
+        /// <param name="rundownSource">
+        /// If given, the rundownSource is used to initialize module and method information.
+        /// This only makes sense in realtime sessions when you need to resolve function names.
+        /// </param>
+        public static TraceLogEventSource CreateFromEventPipeEventSource(EventPipeEventSource source, EventPipeSession rundownSource = null)
```

Comment:
I think that this function should be private.  There is already a path to create a `TraceLog` from a file path, and I would like to make sure that we ensure that someone doesn't pass an `EventPipeEventSource` from a file into this method.